packages/core/src/lib/input/utils.ts

```diff
@@ -12,3 +12,13 @@ export const parseNumericInputValue = (
   type: NumericInputTypes = 'number'
 ) =>
   type === 'number' ? Number.parseFloat(value) : Number.parseInt(value, 10);
+
+export const patternForNumericInput = (
+  type: NumericInputTypes,
+  allowNegative: boolean
+) => {
+  const signPattern = allowNegative ? /[-+]?/u : /\+?/u;
+  return type === 'number'
+    ? new RegExp(`^${signPattern.source}[0-9]+(\\.[0-9]+)?$`, 'u')
+    : new RegExp(`^${signPattern.source}[0-9]+$`, 'u');
```

Comment:
Tried to add comments to these lines, but I cant add comments in the ternary otherwise prettier reformats it and then fails on `line-comment-position`

And I can't break it out into an if-else block because there is also a format lint for prefer-ternary